src/Evolution/Initialization/FunctionOfTimeFromFile.hpp

```diff
@@ -0,0 +1,145 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <tuple>
+#include <unordered_map>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Domain/FunctionsOfTime/Tags.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Evolution/Initialization/Tags.hpp"
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/Dat.hpp"
+#include "IO/H5/File.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "ParallelAlgorithms/Initialization/MergeIntoDataBox.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace Initialization {
+namespace Actions {
+/// \ingroup InitializationGroup
+/// \brief Initialize a `FunctionsOfTime` from dat files in an
+/// H5 file.
+///
+/// Uses:
+///  - DataBox:
+///    - `Initialization::Tags::FuncOfTimeFile`
+///    - `Initialization::Tags::FuncOfTimeSetNames`
+///
+/// DataBox changes:
+/// - Adds:
+///   - A `FunctionsOfTime` containing a `FunctionOfTime` for each dat
```

Comment:
I think this should be `:domain::Tags::FunctionsOfTime` since we care about exactly what tags are being added. You can add the description after the tag name

Reply:
Did you mean `::domain::Tags::FunctionsOfTime`?